lib/api/auth.js

```diff
@@ -0,0 +1,23 @@
+const { Base64 } = require('js-base64');
+
+const AUTH_TYPES = {
+  BASIC_AUTH: 'BASIC_AUTH',
+  API_TOKEN: 'API_TOKEN',
+  OAUTH_ACCESS_TOKEN: 'OAUTH_ACCESS_TOKEN',
+};
+
+function createAuthorizationHeaderValue(config) {
+  const { authType, email, password, zendeskAdminToken } = config;
```

Comment:
I think for a nicer interface, we should validate AuthType here.

ie, assertRequired(authType) ... or whatever syntax is for this function.

If authType not in AUTH_TYPE.values() -- please provide one of XXX

Reply:
Very good point, extra validation added!